packages/crypto/src/bls12_318/hash.rs

```diff
@@ -0,0 +1,102 @@
+use ark_bls12_381::{g1, g2};
+use ark_ec::{
+    hashing::{
+        curve_maps::wb::WBMap, map_to_curve_hasher::MapToCurveBasedHasher, HashToCurve as _,
+    },
+    short_weierstrass::Projective,
+};
+use ark_ff::field_hashers::DefaultFieldHasher;
+use ark_serialize::CanonicalSerialize;
+use sha2::Sha256;
+
+use crate::{CryptoError, BLS12_381_G1_POINT_LEN, BLS12_381_G2_POINT_LEN};
+
+type HashToCurve<CurveConfig, Hash> =
+    MapToCurveBasedHasher<Projective<CurveConfig>, DefaultFieldHasher<Hash>, WBMap<CurveConfig>>;
+
+#[derive(Clone, Copy, Debug)]
+#[non_exhaustive]
+pub enum HashFunction {
+    Sha256,
+}
+
+#[doc(hidden)]
+impl HashFunction {
+    pub fn from_u32(idx: u32) -> Result<Self, CryptoError> {
+        let hash = match idx {
+            0 => Self::Sha256,
+            _ => return Err(CryptoError::unknown_hash_function()),
+        };
+
+        Ok(hash)
+    }
+}
+
+pub fn bls12_381_hash_to_g1(
+    hash: HashFunction,
+    msg: &[u8],
+    dst: &[u8],
+) -> [u8; BLS12_381_G1_POINT_LEN] {
+    let point = match hash {
+        HashFunction::Sha256 => HashToCurve::<g1::Config, Sha256>::new(dst)
+            .unwrap()
+            .hash(msg)
+            .unwrap(),
```

Comment:
Could you add a comment why the unwraps are safe (if they are safe to do)?

Reply:
When following the code back through the entangled ~mess~ *crate graph* that the ark dependencies are, there isn't a single point in the configuration we chose where any of the two unwraps could panic.  

But my assumption that this can't panic comes from the IETF doc itself: <https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-12#section-5.4.1>

Since the underlying curve implementation, when implemented sanely, should never request 255 curve elements at the same time, the expansion will always finish without exiting with an error (since that is the only "ABORT" condition).

Therefore we can conclude, if the implementation is done as defined in the IETF publication, won't ever error out.

Reply:
If that's a good enough explanation I can add that as a disclaimer to the top of the module.

Reply:
Yeah that's great, let's add it